src/CSS Gradients/main.vala

```diff
@@ -0,0 +1,117 @@
+#! /usr/bin/env -S vala workbench.vala --pkg libadwaita-1 --pkg gtksourceview-5
+
+private GtkSource.Buffer gtksource_buffer;
+private Adw.ComboRow combo_row_gradient_type;
+private Adw.SpinRow spin_row_angle;
+private Gtk.ColorDialogButton button_color_1;
+private Gtk.ColorDialogButton button_color_2;
+private Gtk.ColorDialogButton button_color_3;
+private Adw.StyleManager style_manager;
+private GtkSource.StyleSchemeManager scheme_manager;
+private Gtk.CssProvider css_provider;
+private string css;
+
+void update () {
+    spin_row_angle.sensitive = combo_row_gradient_type.selected != 1;
+    css = generate_css ();
+    gtksource_buffer.set_text (css, -1);
+    update_css_provider (css);
+}
+
+void update_css_provider (string css) {
+    var display = Gdk.Display.get_default ();
+    if (css_provider == null) {
+        css_provider = new Gtk.CssProvider ();
+    } else {
+        Gtk.StyleContext.remove_provider_for_display (display, css_provider);
+    }
+
+    css_provider.load_from_string (css);
+    Gtk.StyleContext.add_provider_for_display (display, css_provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
+}
+
+string generate_css () {
+    double angle = spin_row_angle.get_value ();
+    string angle_string = angle.to_string ();
+    string first_color_string = button_color_1.get_rgba ().to_string ();
+    string second_color_string = button_color_2.get_rgba ().to_string ();
+    string third_color_string = button_color_3.get_rgba ().to_string ();
+
+    uint selected = combo_row_gradient_type.selected;
+    if (selected == 0) {
+        css = """.background-gradient {
+    background-image: linear-gradient(
+        %sdeg,
+        %s,
+        %s,
+        %s
+    );
+}""".printf (angle_string, first_color_string, second_color_string, third_color_string);
+    } else if (selected == 1) {
+        css = """.background-gradient {
+    background-image: radial-gradient(
+        %s,
+        %s,
+        %s
+    );
+}""".printf (first_color_string, second_color_string, third_color_string);
+    } else if (selected == 2) {
+        css = """.background-gradient {
+    background-image: conic-gradient(
+        from %sdeg,
+        %s,
+        %s,
+        %s
+    );
+}""".printf (angle_string, first_color_string, second_color_string, third_color_string);
+    }
+    return css;
+}
+
+void update_color_scheme () {
+    string scheme_name = style_manager.dark ? "Adwaita-dark" : "Adwaita";
+    var scheme = scheme_manager.get_scheme (scheme_name);
+
+    if (scheme != null) {
+        gtksource_buffer.set_style_scheme (scheme);
```

Comment:
Use properties instead of getters/setters, applies on line 113 too
```suggestion
        gtksource_buffer.style_scheme = scheme;
```

Reply:
I keep missing out on these, I tried removing the setter for style scheme but didnt know you could use equals . Is there a documentation that could tell me or is trial and error the way to go :)

Reply:
Sure! In Valadoc, all properties are listed at the top of the content (for example, the GtkBox documentation: https://valadoc.org/gtk4/Gtk.Box.html)

In case you're working with the C documentation, properties are listed near the end of the content. Also, if you want to check if a method has a property associated to it

Reply:
Thank you! Will look into this for upcoming demos